src/CartesianProduct.jl

```diff
@@ -562,3 +563,143 @@ function same_block_structure(x::AbstractVector{S1}, y::AbstractVector{S2}
     end
     return true
 end
+
+"""
+    block_structure(cpa::CartesianProductArray{N}) where {N}
+
+Returns an array containing the dimension ranges of each block in a `CartesianProductArray`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+
+### Output
+
+A vector of ranges
+
+### Example
+
+```jldoctest
+julia> using LazySets: block_structure
+
+julia> cpa = CartesianProductArray([BallInf(zeros(n), 1.0) for n in [3, 1, 2]]);
+
+julia> block_structure(cpa)
+3-element Array{UnitRange{Int64},1}:
+ 1:3
+ 4:4
+ 5:6
+```
+"""
+function block_structure(cpa::CartesianProductArray{N}) where {N}
+    result = Vector{UnitRange{Int}}(undef, length(array(cpa)))
+    start_index = 1
+    @inbounds for (i, bi) in enumerate(array(cpa))
+        end_index = start_index + dim(bi) - 1
+        result[i] = start_index:end_index
+        start_index = end_index + 1
+    end
+    return result
+end
+
+
+"""
+    block_to_dimension_indices(cpa::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+
+Returns a vector mapping block index `i`
+to tuple `(f, l)` such that either `f = l = -1` or `f` is the first dimension index and `l` is the last dimension
+index of the `i`-th block, depending on whether one of the block's dimension indices is specified in `vars`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+- `vars` -- list containing the variables of interest, sorted in ascending order
+
+### Output
+
+A vector of tuples, where values in tuple relate to range of dimensions in the i-th block.
+
+### Example
+
+```jldoctest
+julia> using LazySets: block_to_dimension_indices
+
+julia> cpa = CartesianProductArray([BallInf(zeros(n), 1.0) for n in [1, 3, 2, 3]]);
+
+julia> block_to_dimension_indices(cpa, [2, 4, 8])
+(Tuple{Int64,Int64}[(-1, -1), (2, 4), (-1, -1), (7, 9)], 2)
+```
+This vector represents the mapping "second block from dimension 2 to dimension 4,
+fourth block from dimension 7 to dimension 9."
+These blocks contain the dimensions specified in `[2, 4, 8]`.
+"""
+function block_to_dimension_indices(cpa::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+    result = fill((-1, -1), length(array(cpa)))
+    non_empty_length = 0
+    start_index, end_index = 1, 0
+    v_i = 1
+    for i in 1:length(cpa.array)
+        end_index += dim(cpa.array[i])
+        if v_i <= length(vars) && vars[v_i] <= end_index
+            result[i] = (start_index, end_index)
+            non_empty_length += 1
+            while v_i <= length(vars) && vars[v_i] <= end_index
+                v_i += 1
+            end
+        end
+        if v_i > length(vars)
+            break
+        end
+        start_index = end_index + 1
+    end
+    return result, non_empty_length
+end
+
+#same but for all variables
+function block_to_dimension_indices(cpa::CartesianProductArray{N}) where {N}
```

Comment:
Do you have an example where it is worth (faster?) using the version with second argument `vars` instead of this one?

Also, add the `@inbounds` macro in front of the loop, as in` @inbounds for i in ...`, to eliminate array bounds checking and improve performance (a bit), since all accesses are known to be in bounds.

Reply:
> Do you have an example where it is worth (faster?) using the version with second argument `vars` instead of this one?

We don't need information about all blocks usually, therefore it is useful to use version with `vars` as the second parameter in functions, which call `block_to_dimension_indices`.

Reply:
Ok. I was asking because i benchmarked these two functions in two cases that i created (one small dim and another high dim), and the general one was slightly *faster* in both cases.